pkg/provider/cloud/hetzner/provider.go

```diff
@@ -50,7 +50,15 @@ func (h *hetzner) ValidateCloudSpec(spec kubermaticv1.CloudSpec) error {
 	}
 
 	client := hcloud.NewClient(hcloud.WithToken(hetznerToken))
-	_, _, err = client.ServerType.List(context.Background(), hcloud.ServerTypeListOpts{})
+
+	if spec.Hetzner.Network == "" {
+		// this validates the token
+		_, _, err = client.ServerType.List(context.TODO(), hcloud.ServerTypeListOpts{})
```

Comment:
hcloud doesn't set own timeouts inside `hcloud.NewClient()` or in consequent calls. Better to use context with reasonable timeout.